cmd/controller/controller.go

```diff
@@ -189,12 +191,50 @@ func (c *Controller) unseal(key string) error {
 	}
 
 	_, err = c.sclient.Secrets(ssecret.GetObjectMeta().GetNamespace()).Create(secret)
-	if err != nil && errors.IsAlreadyExists(err) {
-		_, err = c.sclient.Secrets(ssecret.GetObjectMeta().GetNamespace()).Update(secret)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		// Error wasn't already exists so is real error
+		return err
 	}
+
+	// Secret already exists so update it in place with new data/owner reference
+	updatedSecret, err := c.updateSecret(secret)
+	if err != nil {
+		return fmt.Errorf("failed to update existing secret: %s", err)
+	}
+	_, err = c.sclient.Secrets(ssecret.GetObjectMeta().GetNamespace()).Update(updatedSecret)
 	return err
 }
 
+func (c *Controller) updateSecret(newSecret *apiv1.Secret) (*apiv1.Secret, error) {
+	existingSecret, err := c.sclient.Secrets(newSecret.GetObjectMeta().GetNamespace()).Get(newSecret.GetObjectMeta().GetName(), metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to read existing secret: %s", err)
+	}
+	existingSecret.Data = newSecret.Data
```

Comment:
`existingSecret` may have come from a shared cache above, so you need to copy it before modifying it:

`existingSecret = existingSecret.DeepCopy()`

Reply:
Good catch. Fixed.